src/commands/branch.rs

```diff
@@ -338,6 +345,45 @@ fn cmd_branch_list(
         }
     }
 
+    if !args.revisions.is_empty() {
+        // Match against local targets only, which is consistent with "jj git push".
+        fn local_targets(branch_target: &BranchTarget) -> &[CommitId] {
+            if let Some(target) = branch_target.local_target.as_ref() {
+                target.adds()
+            } else {
+                &[]
+            }
+        }
+
+        // Intersects with the set of all branch targets to minimize the lookup space.
+        let filter_expressions: Vec<_> = args
+            .revisions
+            .iter()
+            .map(|revision_str| workspace_command.parse_revset(revision_str))
+            .try_collect()?;
+        let filter_expression = RevsetExpression::union_all(&filter_expressions);
```

Comment:
You can leave this as a TODO, but both here and with `jj log`, it would be nice to warn the user if one of the revsets doesn't match anything.

Reply:
I disagree. I don't think I would want a warning for queries like `-r "description(WIP)"`. And invalid symbol is error.

Reply:
I don't have a very strong opinion at the moment. I think I'd appreciate a warning in the `-r "description(WIP)"` case, but I might be missing something.

Reply:
Well, I just don't wanna see a warning if the revset is expected to be empty, and `log -r 'description(WIP)'` could be such example.